test/test_utils/GenericRequestResponseHandler.h

```diff
@@ -0,0 +1,128 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+// #include "RSocketTests.h"
+#include "yarpl/Single.h"
+
+#include "folly/ExceptionWrapper.h"
+
+using namespace yarpl;
+using namespace yarpl::single;
+using namespace rsocket;
+using namespace rsocket::tests;
+using namespace rsocket::tests::client_server;
+
+namespace rsocket { namespace tests {
+
+using StringPair = std::pair<std::string, std::string>;
+
+std::ostream& operator<<(std::ostream& os, StringPair const& payload) {
+  return os << "('" << payload.first << "', '" << payload.second << "')";
+}
+
+struct ResponseImpl {
+  using Response = std::shared_ptr<ResponseImpl>;
+  using MapFunc = std::function<Response(StringPair const&)>;
+
+  enum Type {
+    P, E, F
```

Comment:
I get this is internal testing code, but `Type::P` is really not descriptive.  I think `PAYLOAD`, `EXN`, and `FUNC` aren't that much more verbose, but definitely more descriptive.

Also `enum class` opportunity.